sdk/tables/Azure.Data.Tables/src/Sas/TableAccountSasBuilder.cs

```diff
@@ -0,0 +1,233 @@
+        /// originates does not match the IP address or address range
+        /// specified on the SAS token, the request is not authenticated.
+        /// When specifying a range of IP addresses, note that the range is
+        /// inclusive.
+        /// </summary>
+        public SasIPRange IPRange { get; set; }
+
+        /// <summary>
+        /// An optional unique value up to 64 characters in length that
+        /// correlates to an access policy specified for the container.
+        /// </summary>
+        public string Identifier { get; set; }
+
+        /// <summary>
+        /// The resource types associated with the shared access signature. The
+        /// user is restricted to operations on the specified resources.
+        /// </summary>
+        public TableAccountSasResourceTypes ResourceTypes { get; set; }
+
+        /// <summary>
+        /// The storage service version to use to authenticate requests made
+        /// with this shared access signature, and the service version to use
+        /// when handling requests made with this shared access signature.
+        /// </summary>
+        internal string Version { get; set; }
+
+        /// <summary>
+        /// Sets the permissions for a table SAS.
+        /// </summary>
+        /// <param name="permissions">
+        /// <see cref="TableAccountSasPermissions"/> containing the allowed permissions.
+        /// </param>
+        public void SetPermissions(TableAccountSasPermissions permissions)
+        {
+            Permissions = permissions.ToPermissionsString();
+        }
+
+        /// <summary>
+        /// Sets the permissions for the SAS using a raw permissions string.
+        /// </summary>
+        /// <param name="rawPermissions">Raw permissions string for the SAS.</param>
+        public void SetPermissions(string rawPermissions)
+        {
+            Permissions = rawPermissions;
+        }
+
+        /// <summary>
+        /// Use an account's <see cref="TableSharedKeyCredential"/> to sign this
+        /// shared access signature values to produce the proper SAS query
+        /// parameters for authenticating requests.
+        /// </summary>
+        /// <param name="sharedKeyCredential">
+        /// The storage account's <see cref="TableSharedKeyCredential"/>.
+        /// </param>
+        /// <returns>
+        /// An instance of <see cref="TableAccountSasQueryParameters"/>.
+        /// </returns>
+        public TableAccountSasQueryParameters ToSasQueryParameters(TableSharedKeyCredential sharedKeyCredential)
+        {
+            sharedKeyCredential = sharedKeyCredential ?? throw Errors.ArgumentNull(nameof(sharedKeyCredential));
+
+            EnsureState();
+
+            var startTime = SasExtensions.FormatTimesForSasSigning(StartsOn);
+            var expiryTime = SasExtensions.FormatTimesForSasSigning(ExpiresOn);
+
+            // String to sign: http://msdn.microsoft.com/en-us/library/azure/dn140255.aspx
```

Comment:
The url should point here - https://docs.microsoft.com/en-us/rest/api/storageservices/create-account-sas#constructing-the-signature-string